tests/test_magpie.py

```diff
@@ -0,0 +1,443 @@
+        self.data["sync_permissions"] = {
+            "user_workspace": {
+                "services": {
+                    "Thredds": {
+                        "Thredds1": [
+                            {"name": self.test_service_name, "type": "service"},
+                            {"name": "private", "type": "directory"},
+                            {"name": "workspace_file1", "type": "file"}],
+                        "Thredds2": [
+                            {"name": self.test_service_name, "type": "service"},
+                            {"name": "workspace_file2", "type": "file"}]}},
+                "permissions_mapping": [
+                    {"Thredds1": ["read"], "Thredds2": ["read"]}]
+            }
+        }
+        with self.cfg_file as f:
+            f.write(yaml.safe_dump(self.data))
+        app = utils.get_test_app(settings={"cowbird.config_path": self.cfg_file.name})
+        # Recreate new magpie service instance with new config
+        ServiceFactory().create_service("Magpie")
+
+        with contextlib.ExitStack() as stack:
+            stack.enter_context(mock.patch("cowbird.services.impl.thredds.Thredds",
+                                           side_effect=utils.MockAnyService))
+            # Create test resources
+            parent_res_id = self.create_test_resource("private", "directory", self.test_service_id)
+            src_res_id = self.create_test_resource("workspace_file1", "file", parent_res_id)
+            target_res_id = self.create_test_resource("workspace_file2", "directory", self.test_service_id)
+
+            data = {
+                "event": ValidOperations.CreateOperation.value,
+                "service_name": "Thredds",
+                "resource_id": str(src_res_id),
+                "resource_full_name": f"/{self.test_service_name}/private/workspace_file1",
+                "name": "read",
+                "access": "allow",
+                "scope": "recursive",
+                "user": self.usr,
+                "group": self.grp
+            }
+
+            # Create permissions
+            resp = utils.test_request(app, "POST", "/webhooks/permissions", json=data)
+            utils.check_response_basic_info(resp, 200, expected_method="POST")
+
+            # Check if permissions were created
+            self.check_user_permissions(target_res_id, ["read", "read-allow-recursive"])
+            self.check_group_permissions(target_res_id, ["read", "read-allow-recursive"])
+
+            # Delete permissions
+            data["event"] = ValidOperations.DeleteOperation.value
+            resp = utils.test_request(app, "POST", "/webhooks/permissions", json=data)
+            utils.check_response_basic_info(resp, 200, expected_method="POST")
+
+            # Check if permissions were deleted
+            self.check_user_permissions(target_res_id, [])
+            self.check_group_permissions(target_res_id, [])
+
+    def test_webhooks_valid_tokens(self):
```

Comment:
Please add a docstring that indicates what is the expected outcome of the test. It is hard to grasp the intension.